storage/gcp/gcp.go

```diff
@@ -178,6 +186,40 @@ func (s *Storage) Get(ctx context.Context, path string) ([]byte, error) {
 	return d, err
 }
 
+// init ensures that the storage represents a log in a valid state.
+func (s *Storage) init(ctx context.Context) error {
+	cpRaw, err := s.Get(ctx, layout.CheckpointPath)
+	if err != nil {
+		if errors.Is(err, gcs.ErrObjectNotExist) {
+			// No checkpoint exists, do a forced (possibly empty) integration to create one
+			cctx, c := context.WithTimeout(ctx, 10*time.Second)
+			defer c()
+			if _, err := s.sequencer.consumeEntries(cctx, DefaultIntegrationSizeLimit, s.integrate, true); err != nil {
+				return fmt.Errorf("forced integrate: %v", err)
+			}
```

Comment:
Why don't we call `updateCP` here?

Reply:
That would not be safe - imagine that two frontends start up, one happens to get a request to `/add` which flushes and then it integrates the entry into the tree and writes an `N=1` checkpoint. Meanwhile, the other frontend starts up, sees that there's no checkpoint, and calls `updateCP` overwriting the `N=1` checkpoint with an `N=0` one.

Someone then has a bad day :)

I've added a comment about safety here.
By going through the full integration flow via the sequencer we're protected against that because it involves a transaction on the `IntCoord` table in Spanner.

Reply:
Gotcha, thanks for adding the comments!